src/Analyzers/API/InvocationConcurrency.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.Build.Experimental;
+
+public enum InvocationConcurrency
```

Comment:
I wouldn't include this. Let's define a reasonable programming model and mandate analyzers to be thread-safe. If really deemed important we can provide a synchronized wrapper/helper for analyzer authors to use if they don't want to think about concurrency.